content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb

```diff
@@ -0,0 +1,64 @@
+#
+# Description: provide the dynamic list content from available credentials
+# for Ansible Provider
+#
+module ManageIQ
+  module Automate
+    module AutomationManagement
+      module AnsibleTower
+        module Operations
+          class AvailableCredentials
+            AUTH_CLASS = "ManageIQ_Providers_AutomationManager_Authentication".freeze
+            def initialize(handle = $evm)
+              @handle = handle
+            end
+
+            def main
+              fill_dialog_field(fetch_list_data)
+            end
+
+            private
+
+            def provider
+              # for provisioning we use     service_template
+              # for reconfig and retire use service
+              service = @handle.root['service_template'] || @handle.root['service']
+              service.try(:job_template, 'Provision').try(:manager)
+            end
+
+            def credentials
+              @handle.vmdb(AUTH_CLASS).where("resource_id = ? AND type = ?",
+                                             provider.try(:id),
+                                             @handle.inputs['credential_type'])
+            end
+
+            def fetch_list_data
+              credential_list = Hash[*credentials.pluck(:id, :name).flatten]
+              @handle.log(:info, "Number of credentials found #{credential_list.keys.count}")
```

Comment:
Good for debugging, but I am not convinced we want to log the count here in final release.